core/src/test/java/com/linecorp/armeria/server/logging/LoggingServiceTest.java

```diff
@@ -154,6 +155,91 @@ public void infoLevel() throws Exception {
                             ", trailers: " + RESPONSE_TRAILERS);
     }
 
+    @Test
+    public void mapRequestLogLevelMapper() throws Exception {
+        when(log.requestHeaders()).thenAnswer(invocation -> RequestHeaders.of(HttpMethod.GET, "/",
+                                                                              "x-req", "test",
+                                                                              "x-res", "test"));
+
+        final LoggingService service =
+                LoggingService.builder()
+                              .requestLogLevelMapper(log -> {
+                                  if (log.requestHeaders().contains("x-req")) {
+                                      return LogLevel.WARN;
+                                  } else {
+                                      return LogLevel.INFO;
+                                  }
+                              })
+                              .responseLogLevelMapper(log -> {
+                                  if (log.requestHeaders().contains("x-res")) {
+                                      return LogLevel.WARN;
+                                  } else {
+                                      return LogLevel.INFO;
+                                  }
+                              })
+                              .newDecorator().apply(delegate);
+
+        service.serve(ctx, REQUEST);
+        verify(logger).warn(REQUEST_FORMAT,
+                            "headers: " + REQUEST_HEADERS + ", content: " + REQUEST_CONTENT +
+                            ", trailers: " + REQUEST_TRAILERS);
+        verify(logger).warn(RESPONSE_FORMAT,
+                            "headers: " + RESPONSE_HEADERS + ", content: " + RESPONSE_CONTENT +
+                            ", trailers: " + RESPONSE_TRAILERS);
+
+        when(log.requestHeaders()).thenAnswer(invocation -> RequestHeaders.of(HttpMethod.GET, "/"));
+
+        service.serve(ctx, REQUEST);
+        verify(logger).info(REQUEST_FORMAT,
+                            "headers: " + REQUEST_HEADERS + ", content: " + REQUEST_CONTENT +
+                            ", trailers: " + REQUEST_TRAILERS);
+        verify(logger).info(RESPONSE_FORMAT,
+                            "headers: " + RESPONSE_HEADERS + ", content: " + RESPONSE_CONTENT +
+                            ", trailers: " + RESPONSE_TRAILERS);
+    }
+
+    @Test(expected = IllegalStateException.class)
```

Comment:
nit: Could you use `assertThatThrownBy()` for an exception matching? We usually prefer asserting with `assertThatXXX` of [`assertj`](https://joel-costigliola.github.io/assertj/assertj-core-quick-start.html) :-)

```java
@Test
public void duplicateSetRequestLogLevelAndMapper() throws Exception {
    assertThatThrownBy(() -> {
        LoggingService.builder()
                      .requestLogLevel(LogLevel.INFO)
                      .requestLogLevelMapper(log -> LogLevel.INFO);
    }).isInstanceOf(IllegalStateException.class);
}
```

Reply:
Umm..I already checked `HBaseClientCompatibilityTest` case. I think both case are allowed with any style. Anyway will fix it.

Reply:
Sorry for the confusion. I guess `HBaseClientCompatibilityTest` was made before our current policy. Actually, we now recommend to build test with [JUnit 5](https://github.com/line/armeria/blob/master/CONTRIBUTING.md#use-junit-5-instead-of-junit-4-for-testing). 😉